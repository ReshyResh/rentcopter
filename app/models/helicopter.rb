class Helicopter < ApplicationRecord
  mount_uploader :avatar, AvatarUploader
  belongs_to :user
  has_many :reservations, dependent: :destroy

  validates :name, :description, :model, :image, presence: true
end
