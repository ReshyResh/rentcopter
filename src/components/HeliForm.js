import { useState } from 'react';
import { PropTypes } from 'prop-types';
import { dateDifference, TODAY, MONTHLATER } from '../logic/date';

const HeliForm = (props) => {
  const { price } = props;

  const initialForm = {
    'start-date': TODAY,
    'end-date': TODAY,
  };

  const [form, setForm] = useState(initialForm);

  const handleChange = (e) => {
    setForm({
      ...form,
      [e.target.name]: e.target.value,
    });
  };

  const amount = dateDifference(form['start-date'], form['end-date'], price);

  const messageVerifier = (amount) => {
    if (amount > 0) {
      return `Reservation cost: ${amount} USD`;
    }
    return 'Invalid end-date';
  };

  const classVerifier = (amount) => {
    if (amount > 0) {
      return 'text-center text-green-400 mb-5';
    }
    return 'text-center text-red-400 mb-5';
  };

  return (
    <form className="flex flex-col">
      <div className="flex space-evenly mb-10">
        <label htmlFor="date-start" className="text-center">
          <span>Reservation start-date:</span>
          <input
            type="date"
            id="start-date"
            name="start-date"
            min={TODAY}
            max={MONTHLATER}
            placeholder="from"
            className="bg-gray-600 rounded-full p-2"
            value={form['start-date']}
            onChange={handleChange}
          />
        </label>
        <label htmlFor="date-end" className="text-center">
          <span>Reservation end-date:</span>
          <input
            type="date"
            id="end-date"
            name="end-date"
            min={TODAY}
            max={MONTHLATER}
            placeholder="until"
            className="bg-gray-600 rounded-full p-2"
            value={form['end-date']}
            onChange={handleChange}
          />
        </label>
      </div>
      <span className={classVerifier(amount)}>
        {messageVerifier(amount)}
      </span>
      <button type="submit" className="heli-form-button">RESERVE</button>
    </form>
  );
};

HeliForm.propTypes = {
  price: PropTypes.string.isRequired,
};

export default HeliForm;
