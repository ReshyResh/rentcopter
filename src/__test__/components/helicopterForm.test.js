import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import HeliForm from '../../components/HeliForm';
import helicopter from '../mockHeli';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';


it('Helicopter form component renders correctly', () => {
  const tree = renderer
    .create(<Router>
        <Provider store={store}>
            <HeliForm helicopter={helicopter} />
        </Provider>
    </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});