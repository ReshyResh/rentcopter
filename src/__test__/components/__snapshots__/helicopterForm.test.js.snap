// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helicopter form component renders correctly 1`] = `
<form
  className="flex flex-col"
  id="reserve_form"
>
  <div
    className="flex space-evenly mb-10"
  >
    <label
      className="text-center"
      htmlFor="date-start"
    >
      <span>
        Reservation start-date:
      </span>
      <input
        className="bg-gray-600 rounded-full p-2"
        id="start_date"
        max="2022-02-25"
        min="2022-01-25"
        name="start-date"
        onChange={[Function]}
        placeholder="from"
        type="date"
        value="2022-01-25"
      />
    </label>
    <label
      className="text-center"
      htmlFor="date-end"
    >
      <span>
        Reservation end-date:
      </span>
      <input
        className="bg-gray-600 rounded-full p-2"
        id="end_date"
        max="2022-02-25"
        min="2022-01-25"
        name="end-date"
        onChange={[Function]}
        placeholder="until"
        type="date"
        value="2022-01-25"
      />
    </label>
  </div>
  <span
    className="text-center text-green-400 mb-5"
  >
    Reservation cost: 500 USD
  </span>
  <button
    className="heli-form-button"
    onClick={[Function]}
    type="button"
  >
    RESERVE
  </button>
</form>
`;
